In [5]:
import pandas as pd
import numpy as np

firmo = pd.read_csv("Firmographic_Information-0.csv")
financial = pd.read_csv("Financials_Annual_-0.csv")
firmo.loc[:,["BQ_ID","BQ_COMPANY_NAME","BQ_COMPANY_ADDRESS1_ZIP5","BQ_COMPANY_ADDRESS1_ZIP9"]].loc[firmo["BQ_ID"] == 100002521540]

BQ_ID BQ_COMPANY_NAME  BQ_COMPANY_ADDRESS1_ZIP5  \
199  100002521540     D.H.S. CORP                   77042.0   

    BQ_COMPANY_ADDRESS1_ZIP9  
199               77042-3809

# Création du dataset join, financial & firmographic 

In [6]:
firmo["BQ_COMPANY_ADDRESS1_ZIP5"].isna().sum()

186

### On a 186 valeurs manquantes pour le zip sur 5302 entreprises 

In [7]:
firmo.loc[firmo["BQ_COMPANY_ADDRESS1_ZIP5"].isna()].loc[:,["BQ_COMPANY_ADDRESS1_LINE_1","BQ_COMPANY_ADDRESS1_CITY","BQ_COMPANY_ADDRESS1_STATE"]]

BQ_COMPANY_ADDRESS1_LINE_1 BQ_COMPANY_ADDRESS1_CITY  \
79                                        NaN                Las Vegas   
118                            New Castle, DE                      NaN   
148                            New Castle, DE                      NaN   
149   Care of Citius Pharmaceuticals Cranford                      NaN   
167                                       NaN                      NaN   
...                                       ...                      ...   
4822                           New Castle, DE                      NaN   
4970                                      NaN              Garden City   
5155                                      NaN                      NaN   
5216                                      NaN                      NaN   
5301                                      NaN                      NaN   

     BQ_COMPANY_ADDRESS1_STATE  
79                          NV  
118          No State Reported  
148          No State Reported  
149          No State Reported  
167                         TX  
...                        ...  
4822         No State Reported  
4970                        NY  
5155                        NY  
5216                        CA  
5301                        NY  

[186 rows x 3 columns]

Mais on pourra dans le futur recrée ces valeurs manquantes à partir d'autres colonnes comme address, city ou state. 

In [8]:
financial.loc[:,["BQ_ID","BQ_COMPANY_NAME"]].sample(5)

BQ_ID                    BQ_COMPANY_NAME
43226  100004746843                         KADANT INC
11589  100000678592               ANTERO MIDSTREAM LLC
57779  100007729941             SCHMITT INDUSTRIES INC
50652  100006349773                           OKTA INC
22079  100068121754  ISHARES MORNINGSTAR SMALL-CAP ETF

In [9]:
firmo.loc[:,["BQ_ID","BQ_COMPANY_NAME","BQ_COMPANY_ADDRESS1_ZIP5","BQ_COMPANY_ADDRESS1_ZIP9"]].sample(5)

BQ_ID            BQ_COMPANY_NAME  BQ_COMPANY_ADDRESS1_ZIP5  \
1284  100025164010     DICE THERAPEUTICS INC.                   94080.0   
3801  100007480451                   ROKU INC                   95112.0   
3004  100005450951  MANITEX INTERNATIONAL INC                   60455.0   
4410  100006140098             NEWMARKET CORP                   23219.0   
1024  100001342015                BOXED, INC.                       NaN   

     BQ_COMPANY_ADDRESS1_ZIP9  
1284               94080-6212  
3801               95112-1030  
3004               60455-2304  
4410               23219-4350  
1024                      NaN

In [10]:
df = financial.set_index("BQ_ID").join(firmo.loc[:,["BQ_ID","BQ_COMPANY_NAME","BQ_COMPANY_ADDRESS1_ZIP5","BQ_COMPANY_ADDRESS1_ZIP9"]].set_index("BQ_ID"), rsuffix="firmo").reset_index()
df.sample()

BQ_ID BQ_COMPANY_NAME             BQ_WEBSITE  BQ_YEAR  \
15594  100067981072    VERADIGM INC  https://veradigm.com/     2014   

         BQ_REVENUE    BQ_EBITDA       BQ_COR  BQ_NET_INCOME  BQ_GROSS_PROFIT  \
15594  1.311931e+09  164751875.0  636806804.0     65451411.0      675124519.0   

       BQ_TOTAL_ASSETS  ...  BQ_TOTAL_DEBT  BQ_TOTAL_LIABILITIES  \
15594      916925967.0  ...            NaN          1.180110e+09   

       BQ_SHAREHOLDERS_EQUITY  BQ_TOTAL_LIABILITIES_AND_EQUITY  \
15594            1.284220e+09                     1.180110e+09   

       BQ_CURRENT_RATIO  BQ_QUICK_RATIO  BQ_DEBT_TO_EQUITY_RATIO  \
15594          0.934437        0.738047                  0.41986   

       BQ_COMPANY_NAMEfirmo  BQ_COMPANY_ADDRESS1_ZIP5  \
15594          VERADIGM INC                   27609.0   

       BQ_COMPANY_ADDRESS1_ZIP9  
15594                27609-5177  

[1 rows x 68 columns]

## Les colonnes qui vont nous intéresser sont :

In [11]:
df.columns

Index(['BQ_ID', 'BQ_COMPANY_NAME', 'BQ_WEBSITE', 'BQ_YEAR', 'BQ_REVENUE',
       'BQ_EBITDA', 'BQ_COR', 'BQ_NET_INCOME', 'BQ_GROSS_PROFIT',
       'BQ_TOTAL_ASSETS', 'BQ_OPERATING_EXPENSES', 'BQ_OPERATING_INCOME',
       'BQ_TAX_AND_INTEREST', 'BQ_GROSS_PROFIT_MARGIN', 'BQ_EBITDA_MARGIN',
       'BQ_ASSET_TURNOVER', 'BQ_NET_PROFIT_MARGIN', 'BQ_RETURN_ON_ASSETS',
       'BQ_RETURN_ON_SALES', 'BQ_REVENUE_PER_EMP', 'BQ_EBITDA_PER_EMP',
       'BQ_COR_PER_EMP', 'BQ_NET_INCOME_PER_EMP', 'BQ_GROSS_PROFIT_PER_EMP',
       'BQ_TOTAL_ASSETS_PER_EMP', 'BQ_PAYROLL_PER_EMP',
       'BQ_OPERATING_EXPENSES_PER_EMP', 'BQ_OPERATING_INCOME_PER_EMP',
       'BQ_TAX_AND_INTEREST_PER_EMP', 'BQ_FISCAL_YEAR_END',
       'BQ_CURRENT_ASSETS', 'BQ_CASH',
       'BQ_TRADE_NOTES_AND_ACCOUNTS_RECEIVABLE',
       'BQ_LESS_ALLOWANCE_FOR_BAD_DEBTS', 'BQ_INVENTORIES',
       'BQ_US_GOVERNMENT_OBLIGATIONS', 'BQ_TAX_EXEMPT_SECURITIES',
       'BQ_OTHER_CURRENT_ASSETS', 'BQ_NON_CURRENT_ASSETS',
       'BQ_LOANS_TO_SHARE

In [22]:
features = ['BQ_COMPANY_NAME', 'BQ_YEAR','BQ_TOTAL_ASSETS','BQ_CURRENT_ASSETS', 'BQ_OTHER_NON_CURRENT_ASSETS' 
            ,'BQ_CASH', 'BQ_US_GOVERNMENT_OBLIGATIONS',
            'BQ_BUILDINGS_AND_OTHER_DEPRECIABLE_ASSETS', 'BQ_LAND',"BQ_DEPLETABLE_ASSETS",
            'BQ_INTANGIBLE_ASSETS_AMORTIZABLE','BQ_TOTAL_DEBT', 'BQ_MORTGAGE_AND_REAL_ESTATE_LOANS']

In [28]:
df[features].loc[df["BQ_COMPANY_NAME"] == "MICROSOFT CORP"]

BQ_COMPANY_NAME  BQ_YEAR  BQ_TOTAL_ASSETS  BQ_CURRENT_ASSETS  \
47310  MICROSOFT CORP     2024     4.842750e+11                NaN   
47311  MICROSOFT CORP     2023     4.705580e+11                NaN   
47312  MICROSOFT CORP     2022     4.119760e+11       1.842570e+11   
47313  MICROSOFT CORP     2021     3.648400e+11       1.696840e+11   
47314  MICROSOFT CORP     2020     3.337790e+11       1.844060e+11   
47315  MICROSOFT CORP     2019     3.013110e+11       1.819150e+11   
47316  MICROSOFT CORP     2018     2.865560e+11       1.755520e+11   
47317  MICROSOFT CORP     2017     2.588480e+11       1.696620e+11   
47318  MICROSOFT CORP     2016     2.503120e+11       1.626960e+11   
47319  MICROSOFT CORP     2015     1.934680e+11       1.396600e+11   
47320  MICROSOFT CORP     2014     1.744720e+11       1.227970e+11   
47321  MICROSOFT CORP     2013     1.723840e+11       1.142460e+11   
47322  MICROSOFT CORP     2012     1.424310e+11       1.014660e+11   
47323  MICROSOFT CORP     2011     1.212710e+11       8.508400e+10   
47324  MICROSOFT CORP     2010     1.087040e+11       7.491800e+10   

       BQ_OTHER_NON_CURRENT_ASSETS       BQ_CASH  \
47310                          NaN           NaN   
47311                          NaN           NaN   
47312                          NaN  3.470400e+10   
47313                          NaN  1.393100e+10   
47314                          NaN  1.422400e+10   
47315                          NaN  1.357600e+10   
47316                          NaN  1.135600e+10   
47317                          NaN  1.194600e+10   
47318                          NaN  7.663000e+09   
47319                          NaN  6.510000e+09   
47320                          NaN  5.595000e+09   
47321                          NaN  8.669000e+09   
47322                          NaN  3.804000e+09   
47323                          NaN  6.938000e+09   
47324                          NaN  9.610000e+09   

       BQ_US_GOVERNMENT_OBLIGATIONS  \
47310                           NaN   
47311                           NaN   
47312                           NaN   
47313                           NaN   
47314                           NaN   
47315                           NaN   
47316                           NaN   
47317                           NaN   
47318                           NaN   
47319                           NaN   
47320                           NaN   
47321                           NaN   
47322                           NaN   
47323                           NaN   
47324                           NaN   

       BQ_BUILDINGS_AND_OTHER_DEPRECIABLE_ASSETS  BQ_LAND  \
47310                                        NaN      NaN   
47311                                        NaN      NaN   
47312                               1.099870e+11      NaN   
47313                               8.754600e+10      NaN   
47314                               7.080300e+10      NaN   
47315                               5.290400e+10      NaN   
47316                               4.385600e+10      NaN   
47317                               3.614600e+10      NaN   
47318                               3.028900e+10      NaN   
47319                               1.835600e+10      NaN   
47320                               1.473100e+10      NaN   
47321                               1.301100e+10      NaN   
47322                               9.991000e+09      NaN   
47323                               8.269000e+09      NaN   
47324                               8.162000e+09      NaN   

       BQ_DEPLETABLE_ASSETS  BQ_INTANGIBLE_ASSETS_AMORTIZABLE  BQ_TOTAL_DEBT  \
47310                   NaN                               NaN            NaN   
47311                   NaN                               NaN            NaN   
47312                   NaN                      7.725200e+10            NaN   
47313                   NaN                      7.882200e+10            NaN   
47314                   NaN                      5

'BQ_OTHER_NON_CURRENT_ASSETS' : 66 000 valeurs manquantes sur 66283\
'BQ_US_GOVERNMENT_OBLIGATIONS' : 66233 valeurs manquantes sur 66283\
BQ_LAND : 66233 valeurs manquantes sur 66283\
BQ_DEPLETABLE_ASSETS : 66233 valeurs manquantes sur 66283\
BQ_TOTAL_DEBT : 52580 valeurs manquantes sur 662283 \
BQ_MORTGAGE_AND_REAL_ESTATE_LOANS : 66233 valeurs manquantes sur 66283

In [103]:
df[features].isna().sum().sort_values()

BQ_COMPANY_NAME                                  0
BQ_YEAR                                          0
BQ_TOTAL_ASSETS                               1417
BQ_CURRENT_ASSETS                            12795
BQ_CASH                                      12990
BQ_BUILDINGS_AND_OTHER_DEPRECIABLE_ASSETS    17603
BQ_INTANGIBLE_ASSETS_AMORTIZABLE             39628
BQ_TOTAL_DEBT                                52580
BQ_OTHER_NON_CURRENT_ASSETS                  65999
BQ_US_GOVERNMENT_OBLIGATIONS                 66233
BQ_LAND                                      66233
BQ_DEPLETABLE_ASSETS                         66233
BQ_MORTGAGE_AND_REAL_ESTATE_LOANS            66233
dtype: int64

<style>
r { color: Red }
o { color: Orange }
g { color: Green }
</style>

#### <o> Pas mal de valeurs manquantes : idée -> créer des variables agrégés TANGIBLE (immobilier, inventaire, mobilier ..) vs INTANGIBLE (stock, cash, debt..) <\o>
<g> 1ère solution : CASH, CURRENT_ASSET vs BUILDING (en droppant les lignes où building est NA)


In [144]:
aggreg = df.loc[:,['BQ_COMPANY_NAME', 'BQ_YEAR',"BQ_CURRENT_ASSETS","BQ_CASH","BQ_BUILDINGS_AND_OTHER_DEPRECIABLE_ASSETS", 'BQ_COMPANY_ADDRESS1_ZIP5']].dropna(how='all', subset=['BQ_BUILDINGS_AND_OTHER_DEPRECIABLE_ASSETS'])
aggreg.sample(5)

BQ_COMPANY_NAME  BQ_YEAR  BQ_CURRENT_ASSETS  \
9860                    ALBIREO PHARMA, INC.     2018        167650000.0   
22342  HOME FEDERAL BANCORP INC OF LOUISIANA     2022        138743008.0   
44421        LEGACY EDUCATION ALLIANCE, INC.     2017         25059000.0   
61022        SUPERIOR GROUP OF COMPANIES INC     2015        103999000.0   
41169              IMMUNE THERAPEUTICS, INC.     2019             4925.0   

           BQ_CASH  BQ_BUILDINGS_AND_OTHER_DEPRECIABLE_ASSETS  \
9860   163885000.0                                  1387000.0   
22342   24765000.0                                 17390000.0   
44421    6005000.0                                  1187000.0   
61022    1036000.0                                 22524000.0   
41169       4925.0                                      691.0   

       BQ_COMPANY_ADDRESS1_ZIP5  
9860                     2142.0  
22342                   71105.0  
44421                   33909.0  
61022                   33772.0  
41169                   32801.0

Recall : **Current Assets** = nnual time series based current assets value which accounts for all company-owned assets that can be converted to cash within one year. Current assets include cash, cash equivalents, accounts receivable, stock inventory, marketable securities, pre-paid liabilities, and other liquid assets.

In [142]:
aggreg.loc[aggreg["BQ_COMPANY_ADDRESS1_ZIP5"] == 1001.0]

Empty DataFrame
Columns: [BQ_COMPANY_NAME, BQ_YEAR, BQ_CURRENT_ASSETS, BQ_CASH, BQ_BUILDINGS_AND_OTHER_DEPRECIABLE_ASSETS, BQ_COMPANY_ADDRESS1_ZIP5]
Index: []

# Joindre les données climatiques via le ZIP code

In [149]:
climate = pd.read_csv("U_S_Climate_Risk_Data-0.csv")
climate

STATE    ZIP  HEAT_RATING  HEAT_THRESHOLD_F  \
0      Massachusetts   1001           65              91.5   
1      Massachusetts   1002           60              89.6   
2      Massachusetts   1003           61              90.3   
3      Massachusetts   1005           54              87.6   
4      Massachusetts   1007           61              89.4   
...              ...    ...          ...               ...   
33295     Washington  99363           46              97.7   
33296     Washington  99371           45              97.8   
33297     Washington  99401           45              92.2   
33298     Washington  99402           48              94.3   
33299     Washington  99403           50              96.0   

       HEAT_WETBULB_THRESHOLD_F  HEAT_MAGNITUDE_1970_F  HEAT_MAGNITUDE_1975_F  \
0                          73.4                   90.4                   90.4   
1                          71.3                   88.6                   88.6   
2                          71.8                   89.2                   89.2   
3                          69.7                   86.4                   86.5   
4                          71.2                   88.3                   88.3   
...                         ...                    ...                    ...   
33295                      67.6                   96.0                   96.4   
33296                      65.7                   96.1                   96.6   
33297                      66.3                   90.6                   91.1   
33298                      67.2                   92.7                   93.1   
33299                      68.0                   94.4                   94.8   

       HEAT_MAGNITUDE_1980_F  HEAT_MAGNITUDE_1985_F  HEAT_MAGNITUDE_1990_F  \
0                       90.8                   91.0                   91.1   
1                       89.0                   89.2                   89.3   
2                       89.6                   89.9                   89.9   
3                       86.8                   87.1                   87.2   
4                       88.7                   89.0                   89.0   
...                      ...                    ...                    ...   
33295                   96.8                   97.1                   97.2   
33296                   96.9                   97.2                   97.2   
33297                   91.3                   91.6                   91.6   
33298                   93.4                   93.7                   93.7   
33299                   95.1                   95.4                   95.4   

       ...  FLOOD_TIDAL_EVENTS_2070_P75  FLOOD_TIDAL_EVENTS_2080  \
0      ...                            0                        0   
1      ...                            0                        0   
2      ...                            0                        0   
3      ...                            0                        0   
4      ...                            0                        0   
...    ...                          ...                      ...   
33295  ...                            0                        0   
33296  ...                            0                        0   
33297  ...                            0                        0   
33298  ...                            0                        0   
33299  ...                            0                        0   

       FLOOD_TIDAL_EVENTS_2080_P25  FLOOD_TIDAL_EVENTS_2080_P75  \
0                                0                            0   
1                                0                            0   
2                                0                            0   
3                                0                            0   
4                                0                            0   
...                            ...                          ...   
33295                            0                            0   
33296  

In [159]:
climate.loc[climate["ZIP"] == 68180]

Empty DataFrame
Columns: [STATE, ZIP, HEAT_RATING, HEAT_THRESHOLD_F, HEAT_WETBULB_THRESHOLD_F, HEAT_MAGNITUDE_1970_F, HEAT_MAGNITUDE_1975_F, HEAT_MAGNITUDE_1980_F, HEAT_MAGNITUDE_1985_F, HEAT_MAGNITUDE_1990_F, HEAT_MAGNITUDE_1995_F, HEAT_MAGNITUDE_2000_F, HEAT_MAGNITUDE_2005_RCP45_F, HEAT_MAGNITUDE_2010_RCP45_F, HEAT_MAGNITUDE_2015_RCP45_F, HEAT_MAGNITUDE_2020_RCP45_F, HEAT_MAGNITUDE_2025_RCP45_F, HEAT_MAGNITUDE_2030_RCP45_F, HEAT_MAGNITUDE_2035_RCP45_F, HEAT_MAGNITUDE_2040_RCP45_F, HEAT_MAGNITUDE_2045_RCP45_F, HEAT_MAGNITUDE_2050_RCP45_F, HEAT_MAGNITUDE_2055_RCP45_F, HEAT_MAGNITUDE_2060_RCP45_F, HEAT_MAGNITUDE_2005_RCP85_F, HEAT_MAGNITUDE_2010_RCP85_F, HEAT_MAGNITUDE_2015_RCP85_F, HEAT_MAGNITUDE_2020_RCP85_F, HEAT_MAGNITUDE_2025_RCP85_F, HEAT_MAGNITUDE_2030_RCP85_F, HEAT_MAGNITUDE_2035_RCP85_F, HEAT_MAGNITUDE_2040_RCP85_F, HEAT_MAGNITUDE_2045_RCP85_F, HEAT_MAGNITUDE_2050_RCP85_F, HEAT_MAGNITUDE_2055_RCP85_F, HEAT_MAGNITUDE_2060_RCP85_F, HEAT_EVENTS_1970, HEAT_EVENTS_1975, HEAT_EVENTS_1980, HEAT_EVENTS_1985, HEAT_EVENTS_1990, HEAT_EVENTS_1995, HEAT_EVENTS_2000, HEAT_EVENTS_2005_RCP45, HEAT_EVENTS_2010_RCP45, HEAT_EVENTS_2015_RCP45, HEAT_EVENTS_2020_RCP45, HEAT_EVENTS_2025_RCP45, HEAT_EVENTS_2030_RCP45, HEAT_EVENTS_2035_RCP45, HEAT_EVENTS_2040_RCP45, HEAT_EVENTS_2045_RCP45, HEAT_EVENTS_2050_RCP45, HEAT_EVENTS_2055_RCP45, HEAT_EVENTS_2060_RCP45, HEAT_EVENTS_2005_RCP85, HEAT_EVENTS_2010_RCP85, HEAT_EVENTS_2015_RCP85, HEAT_EVENTS_2020_RCP85, HEAT_EVENTS_2025_RCP85, HEAT_EVENTS_2030_RCP85, HEAT_EVENTS_2035_RCP85, HEAT_EVENTS_2040_RCP85, HEAT_EVENTS_2045_RCP85, HEAT_EVENTS_2050_RCP85, HEAT_EVENTS_2055_RCP85, HEAT_EVENTS_2060_RCP85, HEAT_WETBULB_MAGNITUDE_1970_F, HEAT_WETBULB_MAGNITUDE_1975_F, HEAT_WETBULB_MAGNITUDE_1980_F, HEAT_WETBULB_MAGNITUDE_1985_F, HEAT_WETBULB_MAGNITUDE_1990_F, HEAT_WETBULB_MAGNITUDE_1995_F, HEAT_WETBULB_MAGNITUDE_2000_F, HEAT_WETBULB_MAGNITUDE_2005_RCP45_F, HEAT_WETBULB_MAGNITUDE_2010_RCP45_F, HEAT_WETBULB_MAGNITUDE_2015_RCP45_F, HEAT_WETBULB_MAGNITUDE_2020_RCP45_F, HEAT_WETBULB_MAGNITUDE_2025_RCP45_F, HEAT_WETBULB_MAGNITUDE_2030_RCP45_F, HEAT_WETBULB_MAGNITUDE_2035_RCP45_F, HEAT_WETBULB_MAGNITUDE_2040_RCP45_F, HEAT_WETBULB_MAGNITUDE_2045_RCP45_F, HEAT_WETBULB_MAGNITUDE_2050_RCP45_F, HEAT_WETBULB_MAGNITUDE_2055_RCP45_F, HEAT_WETBULB_MAGNITUDE_2060_RCP45_F, HEAT_WETBULB_MAGNITUDE_2005_RCP85_F, HEAT_WETBULB_MAGNITUDE_2010_RCP85_F, HEAT_WETBULB_MAGNITUDE_2015_RCP85_F, HEAT_WETBULB_MAGNITUDE_2020_RCP85_F, HEAT_WETBULB_MAGNITUDE_2025_RCP85_F, HEAT_WETBULB_MAGNITUDE_2030_RCP85_F, HEAT_WETBULB_MAGNITUDE_2035_RCP85_F, HEAT_WETBULB_MAGNITUDE_2040_RCP85_F, HEAT_WETBULB_MAGNITUDE_2045_RCP85_F, HEAT_WETBULB_MAGNITUDE_2050_RCP85_F, HEAT_WETBULB_MAGNITUDE_2055_RCP85_F, HEAT_WETBULB_MAGNITUDE_2060_RCP85_F, HEAT_WETBULB_EVENTS_1970, HEAT_WETBULB_EVENTS_1975, ...]
Index: []

[0 rows x 441 columns]

In [155]:
financ_with_clim = df.set_index("BQ_COMPANY_ADDRESS1_ZIP5").join(climate.set_index("ZIP")).reset_index()
financ_with_clim.sample(5)

BQ_COMPANY_ADDRESS1_ZIP5         BQ_ID            BQ_COMPANY_NAME  \
56305                   90703.0  100007353937         REVOLVE GROUP, INC   
64772                   68179.0  100009141303         UNION PACIFIC CORP   
57937                   66202.0  100007760924              SEABOARD CORP   
6334                    50309.0  100028330798  F&G ANNUITIES & LIFE, INC   
50529                   95051.0  100006332991                NVIDIA CORP   

                          BQ_WEBSITE  BQ_YEAR    BQ_REVENUE     BQ_EBITDA  \
56305       https://www.revolve.com/     2018  4.987390e+08  4.403400e+07   
64772            https://www.up.com/     2013  2.196300e+10  9.351000e+09   
57937  https://www.seaboardcorp.com/     2017  5.809000e+09  5.520000e+08   
6334         https://www.fglife.com/     2019  1.421422e+08  2.762351e+07   
50529        https://www.nvidia.com/     2012  4.280159e+09  8.716600e+08   

             BQ_COR  BQ_NET_INCOME  BQ_GROSS_PROFIT  ...  \
56305  2.334330e+08   3.063800e+07     2.653060e+08  ...   
64772  3.534000e+09   4.388000e+09     1.842900e+10  ...   
57937  5.260000e+09   2.460000e+08     5.490000e+08  ...   
6334   2.356676e+07   1.549143e+07     1.185755e+08  ...   
50529  2.053816e+09   5.625360e+08     2.226343e+09  ...   

       FLOOD_TIDAL_EVENTS_2070_P75  FLOOD_TIDAL_EVENTS_2080  \
56305                          0.0                      0.0   
64772                          NaN                      NaN   
57937                          0.0                      0.0   
6334                           0.0                      0.0   
50529                          0.0                      0.0   

       FLOOD_TIDAL_EVENTS_2080_P25  FLOOD_TIDAL_EVENTS_2080_P75  \
56305                          0.0                          0.0   
64772                          NaN                          NaN   
57937                          0.0                          0.0   
6334                           0.0                          0.0   
50529                          0.0                          0.0   

       FLOOD_TIDAL_EVENTS_2090  FLOOD_TIDAL_EVENTS_2090_P25  \
56305                      0.0                          0.0   
64772                      NaN                          NaN   
57937                      0.0                          0.0   
6334                       0.0                          0.0   
50529                      0.0                          0.0   

       FLOOD_TIDAL_EVENTS_2090_P75  FLOOD_TIDAL_EVENTS_2100  \
56305                          0.0                      0.0   
64772                          NaN                      NaN   
57937                          0.0                      0.0   
6334                           0.0                      0.0   
50529                          0.0                      0.0   

       FLOOD_TIDAL_EVENTS_2100_P25  FLOOD_TIDAL_EVENTS_2100_P75  
56305                          0.0                          0.0  
64772                          NaN                          NaN  
57937                          0.0                          0.0  
6334                           0.0                          0.0  
50529                          0.0                          0.0  

[5 rows x 508 columns]

# Partie Modélisation